### ISSUE TITLE
Feature Request: add update_many, add_many, and delete_many methods.

### ISSUE THREAD

**Issue by alexstandiford** [berlindb/core#18]
There are many situations where it's practical to create, or update many records at one time, using the same values. Currently, the only way to-do this with Berlin is by querying the records, looping through each one, and updating the record one at a time.

While working with Mongoose, I found that they have two methods, `updateMany`, `addMany`, and `deleteMany`, which would make the database update using a single query instead of x+1 queries as mentioned in the above paragraph.

**Comment by alexstandiford** [comment]
WIP PR for `update_many` method. #20

**Comment by alexstandiford** [comment]
So, after working with BerlinDB for a while, I've come to realize that this is not a great idea, in-general. The query is technically faster than looping through and running many queries, however, in a production site this could really gum up the WordPress install while this query runs.

If anything, I think `update_many` could potentially be basically a method that accepts a query, and will delete all records that match that query. The key difference is it would be little more than a helper function that would use a `foreach` loop to loop through the found records, and delete each one.

**Comment by JJJ** [comment]
The nice thing about having methods be wrappers like you suggest @alexstandiford, is that it offers up an opportunity to build in some kind of batch processing/jobs/deferment type stuff. 👍

**Comment by alexstandiford** [comment]
Right, which is exactly how you would _actually_ delete large quantities of data in WordPress, anyway. These seem like a great opportunity to make a static method.

Something like:

```php

/**
 * @var array $deleted array of deleted data IDs
**/
$deleted = Query::delete_many( [/** query args **/] );

```

```php

/**
 * @var array $inserted array of inserted data IDs
**/
$inserted = Query::insert_many([/** array of items to insert **/]);

```

```php

/**
 * @var array $inserted array of updated data IDs
**/
$update_many = Query::update_many([/** array of items to update, potentially keyed by the item ID or something like that. **/]);

```

**Comment by JJJ** [comment]
Looking at this from the perspective of the existing `add_item()`, `update_item()`, and `delete_item()` methods, the wrapper approach is definitely the right call. Each of those methods does a lot more than issue SQL — they handle meta, cache invalidation, `transition_item()`, and `do_action()` hooks. A raw bulk SQL query would bypass all of that and leave things in an inconsistent state.

A few thoughts on the API shape:

**Instance methods, not static** — The existing CRUD methods are instance methods, and these should follow suit. Static methods don't fit naturally here because the Query class needs its own internal state (column defs, table name, prefix, etc.) to do its work.

**Consistent return values** — Each method should return an array of item IDs that succeeded, and skip/log failures individually rather than bailing entirely on a partial failure:

```php
// Returns array of inserted IDs (empty on total failure)
$inserted = $query->add_items( [ [...], [...], [...] ] );

// Returns array of updated IDs
$updated = $query->update_items( $query_args, $data );

// Returns array of deleted IDs  
$deleted = $query->delete_items( $query_args );
```

**`update_items` / `delete_items` signature** — These should accept the same query args that `query()` accepts, so callers can express "update/delete everything matching this condition" naturally. Internally they'd call `get_item_ids()` then loop.

**No chunking built-in** — Chunking/batching (e.g. WP Cron, Action Scheduler) is a caller concern. The methods should just be clean loops over the single-item methods. If someone needs to process 100k rows safely, they call `delete_items()` on smaller slices themselves.

The naming question: `add_many`/`update_many`/`delete_many` (Mongoose-style) vs. `add_items`/`update_items`/`delete_items` (consistent with the existing `get_item`/`get_items` pattern). I'd lean toward `*_items` to stay consistent with the existing naming convention.

**Comment by JJJ** [comment]
Closing as delivered. The batch CRUD trio now exists:

- **`delete_items( $set )`** (#214, #130) — delete a set by ID, list, or query-var filter
- **`update_items( $set, $data )`** (#214) — write one set of values to a set
- **`add_items( $rows )`** (this) — insert a list of new-item data arrays

All three are built as `Operations\` verbs (`Base` + `Delete`/`Update`/`Add`) and
**loop the per-item primitive** (`add_item`/`update_item`/`delete_item`) rather than
emitting a single bulk statement. That deliberately preserves per-row defaults,
primary-key/UUID generation, sanitization, meta handling, cache priming/invalidation,
and the CRUD/transition actions — the things a raw multi-row `INSERT`/`UPDATE`/`DELETE`
would silently skip.

`add_items()` returns the new IDs in input order (each slot the new ID, or `false`
where that one insert failed) since the IDs are the point of a batch insert; the other
two return a count. Empty inputs are no-ops (and a filter that compiles to no `WHERE`
never widens to "all rows").

**Note on the original "single query instead of x+1" motivation:** that performance
aspiration is intentionally *not* met here — correctness-first looping is x+1 by design.
A true single-query bulk primitive (Rails-style `insert_all`/`update_all`/`delete_all`
that knowingly bypasses hooks/cache) would be a separate, opt-in feature; will file a
follow-up if there's demand for the raw throughput.

Shipped in 86e65da.
